src/webui/www/private/scripts/client.js

```diff
@@ -672,6 +672,28 @@ window.addEventListener("DOMContentLoaded", () => {
             tracker.classList.toggle("selectedFilter", (Number(tracker.id) === selectedTracker));
     };
 
+    const statusSortOrder = Object.freeze({
+        "unknown": -1,
+        "forcedDL": 0,
+        "downloading": 1,
+        "forcedMetaDL": 2,
+        "metaDL": 3,
+        "stalledDL": 4,
+        "forcedUP": 5,
+        "uploading": 6,
+        "stalledUP": 7,
+        "checkingResumeData": 8,
+        "queuedDL": 9,
+        "queuedUP": 10,
+        "checkingUP": 11,
+        "checkingDL": 12,
+        "stoppedDL": 13,
+        "stoppedUP": 14,
+        "moving": 15,
+        "missingFiles": 16,
+        "error": 17
+    });
```

Comment:
Switch this from an object to a Map? I think that would also guarantee that the performance would be fine to do away with the `_statusOrder` property, but will leave that up to you.

```suggestion
    const statusSortOrder = new Map(Object.entries({
        "unknown": -1,
        "forcedDL": 0,
        "downloading": 1,
        "forcedMetaDL": 2,
        "metaDL": 3,
        "stalledDL": 4,
        "forcedUP": 5,
        "uploading": 6,
        "stalledUP": 7,
        "checkingResumeData": 8,
        "queuedDL": 9,
        "queuedUP": 10,
        "checkingUP": 11,
        "checkingDL": 12,
        "stoppedDL": 13,
        "stoppedUP": 14,
        "moving": 15,
        "missingFiles": 16,
        "error": 17
    }));
```

Reply:
I think the approved version is fine. 
I quite like it being immutable. The previous one is inferior because it does more work than necessary when comparing.
As for Map, Object performance difference I don't think it matters because now it is only relevant when the state updates which is fewer than the comparison function runs.